tardis/analysis.py

```diff
@@ -77,13 +77,13 @@ def update_last_interaction_filter(self):
             packet_filter = (self.last_line_interaction_angstrom > self.wavelength_start) & \
                       (self.last_line_interaction_angstrom < self.wavelength_end)
         elif self.packet_filter_mode == 'line_in_nu':
-            line_in_nu = self.lines.wavelength.ix[self.last_line_interaction_in_id].values
+            line_in_nu = self.lines.wavelength.iloc[self.last_line_interaction_in_id].values
             packet_filter = (line_in_nu > self.wavelength_start.to(u.angstrom).value) & \
                 (line_in_nu < self.wavelength_end.to(u.angstrom).value)
 
                 
-        self.last_line_in = self.lines.ix[self.last_line_interaction_in_id[packet_filter]]
-        self.last_line_out = self.lines.ix[self.last_line_interaction_out_id[packet_filter]]
+        self.last_line_in = self.lines.iloc[self.last_line_interaction_in_id[packet_filter]]
+        self.last_line_out = self.lines.iloc[self.last_line_interaction_out_id[packet_filter]]
 
```

Comment:
Actually, we have to use ``ix`` since we want to get the line with a certain ID back and the lines dataframe is indexed by these IDs. The reason why in your trials ``idx`` did not work but ``iloc`` did is because you did not properly mask out the non-interacting or purely e-scattering packets. Thus, many packets have a ``-1`` as line ID (which simply means *no line interaction performed*). ``ix`` fails in this case, since there is no line with ID -1 in the data frame. ``iloc`` simply returns the last element in the data frame (some line with a very long wavelength). But of course, this is not a line which contributes to the wavelength window in question.

Once a proper masking is implemented, ``ix`` can be safely used again